src/imageproxy.rs

```diff
@@ -220,10 +226,20 @@ impl ImageProxy {
                 supported
             ));
         }
+        r.protover = protover;
 
         Ok(r)
     }
 
+    /// If the proxy supports at least semver 0.2.3, return a wrapper with methods from that version.
```

Comment:
Interesting approach. I can imagine this becoming unwieldy if we have lots of different versions we need to expose. But I guess clients are coupled enough that we can ensure there's not a large skew to maintain.